test-functional/src/test/java/com/vmware/test/functional/saas/environment/CalculateEnvironmentUtilsTest.java

```diff
@@ -25,6 +29,8 @@
  * Tests to verify how {@link LocalTestProcessCtl} environment is calculated.
  */
 @TestPropertySource(properties = { "prefix.app.env.var1=newValue1", "app.env.var2=newValue2" })
+@ContextHierarchy(@ContextConfiguration(classes = SharedConfig.class))
+@FunctionalTest
```

Comment:
@FunctionalTest is necessary here, because, after the spring version bump that kicks in the changes related to the improvements of the code that discovers auto configs via spring.factories, we start seeing this error fired by TestClassScopeExecutionListener:

```
java.lang.IllegalStateException: Scope testClass not found.
```

It is strange that we didn't see tat before - TestClassScopeExecutionListener is registered globally (for all spring tests), while registration of the scope happens in FunctionalConfig and thus is applicable only for @FunctionalTest annotated test classes